modules/data/tools/smart_recorder/emergency_mode_trigger_beta.cc

```diff
@@ -0,0 +1,97 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/data/tools/smart_recorder/emergency_mode_trigger_beta.h"
+
+#include <memory>
+
+#include "cyber/common/log.h"
+#include "modules/common/adapters/adapter_gflags.h"
+#include "modules/control/proto/control_cmd.pb.h"
+
+namespace apollo {
+namespace data {
+
+using apollo::canbus::Chassis;
+
+constexpr int ITEM_SIZE = 10;
+constexpr float MAX_DELTA = 10.0f;
+
+EmergencyModeTriggerBeta::EmergencyModeTriggerBeta() {
+  trigger_name_ = "EmergencyModeTriggerBeta";
+}
+
+void EmergencyModeTriggerBeta::Pull(const RecordMessage& msg) {
+  if (!trigger_obj_->enabled()) {
+    return;
+  }
+  std::shared_ptr<EmergencyMessage> emergency_msg =
+    std::make_shared<EmergencyMessage>();
+  emergency_msg->Initial(msg);
+
+  if (msg.channel_name == FLAGS_chassis_topic) {
```

Comment:
Do not need to check emergency here, since it's already taken care of in "emergency_mode_trigger"